lib/linux_admin/network_interface.rb

```diff
@@ -0,0 +1,183 @@
+require 'ipaddr'
+
+module LinuxAdmin
+  class NetworkInterface
+    include Common
+
+    # Cached class instance variable for what distro we are running on
+    @dist_class = nil
+
+    # Gets the subclass specific to the local Linux distro
+    #
+    # @param clear_cache [Boolean] Determines if the cached value will be reevaluated
+    # @return [Class] The proper class to be used
+    def self.dist_class(clear_cache = false)
+      @dist_class = nil if clear_cache
+      @dist_class ||= begin
+        if [Distros.rhel, Distros.fedora].include?(Distros.local)
+          NetworkInterfaceRH
+        else
+          NetworkInterfaceGeneric
+        end
+      end
+    end
+
+    class << self
+      private
+
+      alias_method :orig_new, :new
+    end
+
+    # Creates an instance of the correct NetworkInterface subclass for the local distro
+    def self.new(*args)
+      if self == LinuxAdmin::NetworkInterface
+        dist_class.new(*args)
+      else
+        orig_new(*args)
+      end
+    end
+
+    # @return [String] the interface for networking operations
+    attr_reader :interface
```

Comment:
For an attr, you don't need the `@return [String]` syntax for YARD (I don't think)